internals/src/lib.rs

```diff
@@ -35,3 +35,30 @@ mod parse;
 #[cfg(feature = "serde")]
 #[macro_use]
 pub mod serde;
+
+/// A conversion trait for unsigned integer types smaller than or equal to 64-bits.
+///
+/// This trait exists because [`usize`] doesn't implement `Into<u64>`. We only support 32 and 64 bit
+/// architectures because of consensus code so we can infallibly do the conversion.
+///
+/// **This trait should not appear in any public APIs because the `internals` crate will never stabalize.**
```

Comment:
This is true of any type or trait in the crate so not sure why this is needed.

BTW there's a WIP Rust feature to detect types from private dependencies in public interface. IIRC it can detect use of the trait already. Maybe we should try to use it regardless of its incompleteness.

Reply:
Both good points, I"ll re-spin. Thanks

Reply:
Did you mean this one https://doc.rust-lang.org/stable/nightly-rustc/rustc_lint_defs/builtin/static.PRIVATE_INTERFACES.html

It doesn't fire if we add it along with this code to `bitcoin/src/lib.rs`
```rust
#![deny(private_interfaces)]

/// REMOVE ME
pub fn bar(_: impl internals::ToU64) {}
```

Reply:
No, that does something else. The feature is in nightly and requires marking dependencies as `public = true` in `Cargo.toml` or not. So we will definitely need a script to inject those into our `Cargo.toml` when linting based on a list defined in `[metadata]` or externally. Also there are some compiler switches I haven't figured out completely yet.

Reply:
Lets leave that until we are closer to `v1.0`s, I'd definitely like to use it then to help catch us.